docs/reference/commandline/build.md

```diff
@@ -483,13 +483,37 @@ image with the content of the squashed layers. This effectively makes it look
 like all `Dockerfile` commands were created with a single layer. The build
 cache is preserved with this method.
 
-**Note**: using this option means the new image will not be able to take
-advantage of layer sharing with other images and may use significantly more
-space.
+The `--squash` option is an experimental feature, and should not be considered
+stable.
 
-**Note**: using this option you may see significantly more space used due to
-storing two copies of the image, one for the build cache with all the cache
-layers in tact, and one for the squashed version.
+
+Squashing layers can be beneficial if your Dockerfile produces multiple layers
+modifying the same files, for example, file that are created in one step, and
+removed in another step. For other use-cases, squashing images may actually have
+a negative impact on performance; when pulling an image consisting of multiple
+layers, layers can be pulled in parallel, and allows sharing layers between
+images (saving space).
+
+For most use cases, multi-stage are a better alternative, as they give more
+fine-grained control over your build, and can take advantage of future
+optimizations in the builder. Refer to the [use multi-stage builds](https://docs.docker.com/engine/userguide/eng-image/multistage-build/)
+section in the userguide for more information.
+
+
+#### Known limitations
+
+The `--squash` option has a number of known limitations:
+
+- When squashing layers, the resulting image cannot take advantage of layer 
+  sharing with other images, and may use significantly more space. Sharing the
+  base image is still supported.
+- When using this option you may see significantly more space used due to
+  storing two copies of the image, one for the build cache with all the cache
+  layers in tact, and one for the squashed version.
+- While squashing layers may produce smaller images, it may 
```

Comment:
The end of this sentence is missing.

Reply:
ugh, let me see if it's in my history, haha

Reply:
@ijc updated; I think I had something like that 😅